test/eventsource_test.js

```diff
@@ -562,610 +383,364 @@ describe('HTTP Request', function () {
-      createServer(function (err, server) {
-        if (err) return done(err)
+        await withEventSource(server, async es => {
+          const request1 = await server.nextRequest()
+          assert.equal(request1.path, '/')
 
-        server.on('request', function (req, res) {
-          res.writeHead(status, 'status message', {
-            'Connection': 'Close'
-          })
-          res.end()
+          const request2 = await server.nextRequest()
+          assert.equal(request2.path, redirectSuffix)
         })
+      })
+    })
+
+    it('causes error event when response is ' + status + ' with missing location', async () => {
+      await withServer(async server => {
+        server.byDefault(TestHttpHandlers.respond(status, {'Connection': 'Close'}))
 
-        var es = new EventSource(server.url)
-        es.onerror = function (err) {
-          es.close()
+        await withEventSource(server, async es => {
+          const errors = startErrorQueue(es)
+          const err = await errors.take()
           assert.equal(err.status, status)
-          assert.equal(err.message, 'status message')
-          server.close(done)
-        }
+        })
       })
     })
   });
 
   [401, 403].forEach(function (status) {
-    it('causes error event when response status is ' + status, function (done) {
-      createServer(function (err, server) {
-        if (err) return done(err)
-
-        server.on('request', function (req, res) {
-          res.writeHead(status, 'status message', {'Content-Type': 'text/html'})
-          res.end()
-        })
+    it('causes error event when response status is ' + status, async () => {
+      await withServer(async server => {
+        server.byDefault(TestHttpHandlers.respond(status))
 
-        var es = new EventSource(server.url)
-        es.onerror = function (err) {
-          es.close()
+        await withEventSource(server, async es => {
+          const errors = startErrorQueue(es)
+          const err = await errors.take()
           assert.equal(err.status, status)
-          assert.equal(err.message, 'status message')
-          server.close(done)
-        }
+        })
       })
     })
   })
 })
 
-describe('HTTPS Support', function () {
-  it('uses https for https urls', function (done) {
-    createHttpsServer(function (err, server) {
-      if (err) return done(err)
-
-      server.on('request', writeEvents(['data: hello\n\n']))
-      var es = new EventSource(server.url, {rejectUnauthorized: false})
-      es.onerror = function () {}
+describe('HTTPS Support', () => {
+  it('uses https for https urls', async () => {
+    await withCloseable(TestHttpServers.startSecure, async server => {
+      server.byDefault(writeEvents(['data: hello\n\n']))
 
-      es.onmessage = function (m) {
-        assert.equal('hello', m.data)
-        es.close()
-        server.close(done)
-      }
-    })
-  })
-})
-
-describe('HTTPS Client Certificate Support', function () {
```

Comment:
I removed this test (and the data files for it) because, even though I believe it is possible to make our `js-test-helpers` HTTPS server code work with a client certificate, I think it is redundant in terms of testing this project— it's really just testing built-in functionality of Node's `https` package. That is, the EventSource code doesn't know anything about client certificates, it's just taking whatever was put in the `https` property of the options and passing it straight through to the Node HTTP client.